src/plugins/discover/public/application/main/services/discover_internal_state_container.ts

```diff
@@ -24,7 +24,7 @@ export interface InternalState {
 }
 
 interface InternalStateTransitions {
-  setDataView: (state: InternalState) => (dataView: DataView) => InternalState;
+  setDataView: (state: InternalState) => (dataView: DataView | undefined) => InternalState;
```

Comment:
This is one of the main changes, related to DiscoverInternalStateContainer, which we use at several places in our UI code. 

```TypeScript
export interface InternalState {
  dataView: DataView | undefined;
  savedDataViews: DataViewListItem[];
  adHocDataViews: DataView[];
  expandedDoc: DataTableRecord | undefined;
  customFilters: Filter[];
 }
```

It already allowed the data view to be undefined, but in our code we make sure this is not the case, and we use ` state.dataView!` to say so

`  const dataView = useInternalStateSelector((state) => state.dataView!);`

Now it's honest, and cleaner, the actually selected data view can be undefined, and Discover doesn't break